src/fields/field.impl.hh

```diff
@@ -771,9 +771,9 @@ void Field<spacedim, Value>::cache_reallocate(const ElementCacheMap &cache_map)
 
 
 template<int spacedim, class Value>
-void Field<spacedim, Value>::cache_update(ElementCacheMap &cache_map, unsigned int i_reg) const {
-    if (region_fields_[i_reg] != nullptr) // skips bounadry regions for bulk fields and vice versa
-        region_fields_[i_reg]->cache_update(value_cache_, cache_map, i_reg);
+void Field<spacedim, Value>::cache_update(ElementCacheMap &cache_map, unsigned int region_idx, unsigned int region_patch_idx) const {
+    if (region_fields_[region_idx] != nullptr) // skips bounadry regions for bulk fields and vice versa
```

Comment:
Move obtaining of region_idx from region_patch_idx here (appropriate method of cache_map)